compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1118,53 +1123,54 @@ private STNode parseFuncDefOrFuncTypeDesc(STNode metadata, STNode visibilityQual
-                endContext();
-                return funcDef;
+                return parseFunctionDefinition(metadata, visibilityQualifier, resourcePath, qualifiers, functionKeyword,
+                                               isObjectMember, isObjectTypeDesc, name);
             case EQUAL_TOKEN:
             case SEMICOLON_TOKEN:
                 endContext();
-                return parseVarDeclWithFunctionType(metadata, visibilityQualifier, qualifiers, functionKeyword,
-                                                    STNodeFactory.createEmptyNode(), isObjectMember, isObjectTypeDesc,
-                                                    false);
+                functionKeyword = validateVisibilityQualifiersOfFunctionType(visibilityQualifier, qualifiers,
+                                                                             functionKeyword, isObjectMember);
+                List<STNode> extractQualifiersList = extractQualifiersOfFunctionType(qualifiers, isObjectMember,
+                                                                                     isObjectTypeDesc);
+                STNode qualifierNodeList = createFuncTypeQualNodeList(qualifiers, false);
+                STNode typeDesc = STNodeFactory.createFunctionTypeDescriptorNode(qualifierNodeList, functionKeyword,
+                                                                                 STNodeFactory.createEmptyNode());
+                // Check if it is a complex type desc starting with function type.
+                typeDesc = parseComplexTypeDescriptor(typeDesc, ParserRuleContext.TOP_LEVEL_FUNC_DEF_OR_FUNC_TYPE_DESC,
+                                                      false);
+                if (isObjectMember) {
+                    STNode objectFieldQualNodeList = STNodeFactory.createNodeList(extractQualifiersList);
+                    return parseObjectFieldRhs(metadata, visibilityQualifier, objectFieldQualNodeList, typeDesc, name,
+                                               isObjectTypeDesc);
+                }
+                startContext(ParserRuleContext.VAR_DECL_STMT);
+                STNode funcTypeName = STNodeFactory.createSimpleNameReferenceNode(name);
+                STNode bindingPattern = createCaptureOrWildcardBP(((STSimpleNameReferenceNode) funcTypeName).name);
+                STNode typedBindingPattern = STNodeFactory.createTypedBindingPatternNode(typeDesc, bindingPattern);
+                return parseVarDeclRhs(metadata, extractQualifiersList, typedBindingPattern, true);
             default:
-                STToken functionName = consume();
-                STToken token = peek();
-                recover(token, ParserRuleContext.OPTIONAL_RELATIVE_PATH, metadata, visibilityQualifier, qualifiers,
-                        functionKeyword, isObjectMember, isObjectTypeDesc);
-                resourcePath = parseOptionalRelativePath(isObjectMember);
```

Comment:
`parseOptionalRelativePath(boolean isObjectMember)` method is no longer used can we remove that?

Reply:
Done.